django/db/migrations/operations/fields.py

```diff
@@ -134,7 +134,7 @@ def reduce(self, operation, app_label):
         ):
             if isinstance(operation, AlterField):
                 return [
-                    AddField(
+                    self.__class__(
```

Comment:
Are these done in order to support a particular third-party application use case? I don't see any tests to make sure we don't regress here though.